File: FIRST-AND-LAST-POSITION-IN-ARRAY/flpa.go
func searchRange(nums []int, target int) []int {
    result := []int{-1, -1}

    // Find first occurrence
    start, end := 0, len(nums)-1
    for start <= end {
        mid := start + (end-start)/2
        if nums[mid] < target {
            start = mid + 1
        } else {
            if nums[mid] == target {
                result[0] = mid
            }
            end = mid - 1
        }
    }

    // Find last occurrence
    start, end = 0, len(nums)-1
    for start <= end {
        mid := start + (end-start)/2
        if nums[mid] > target {
            end = mid - 1
        } else {
            if nums[mid] == target {
                result[1] = mid
            }
            start = mid + 1
        }
    }

    return result
}

File: FIRST-AND-LAST-POSITION-IN-ARRAY/flpa.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] result = new int[2];
        
        // Find first occurrence
        int start = 0, end = nums.length - 1;
        int res = -1;
        while (start <= end) {
            int mid = start + (end - start) / 2;
            if (nums[mid] < target) {
                start = mid + 1;
            } else {
                if (nums[mid] == target) {
                    res = mid;
                }
                end = mid - 1;
            }
        }
        result[0] = res;

        // Find last occurrence
        start = 0;
        end = nums.length - 1;
        res = -1;
        while (start <= end) {
            int mid = start + (end - start) / 2;
            if (nums[mid] > target) {
                end = mid - 1;
            } else {
                if (nums[mid] == target) {
                    res = mid;
                }
                start = mid + 1;
            }
        }
        result[1] = res;

        return result;
    }
}
